dipy/reconst/ivim.py

```diff
@@ -0,0 +1,494 @@
+""" Classes and functions for fitting ivim model """
+from __future__ import division, print_function, absolute_import
+
+from distutils.version import LooseVersion
+
+import numpy as np
+import scipy
+import warnings
+from dipy.core.gradients import gradient_table
+from dipy.reconst.base import ReconstModel
+from dipy.reconst.multi_voxel import multi_voxel_fit
+
+SCIPY_LESS_0_17 = LooseVersion(scipy.version.short_version) < '0.17'
+
+if SCIPY_LESS_0_17:
+    leastsq = scipy.optimize.leastsq
+else:
+    least_squares = scipy.optimize.least_squares
```

Comment:
I would change these lines to: 

    if SCIPY_LESS_0_17:
        from scipy.optimize import leastsq
    else:
        from scipy.optimize import least_squares